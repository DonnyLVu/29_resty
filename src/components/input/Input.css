.Input {
  padding: 20px;
  width: 60%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}