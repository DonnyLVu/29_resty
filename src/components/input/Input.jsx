import React from 'react';
import PropTypes from 'prop-types';

const Input = ({ url, method, body, onSubmit, onChange }) => {
  return (
    <>
      <form onSubmit={onSubmit}>
        <input type="text"
          name="url"
          placeholder="This is the URL"
          value={url}
          onChange={onChange} />

        <div>
          <label data-testid="get">GET</label>
          <input type="radio"
            name="method"
            value="GET"
            checked={method === 'GET'}
            onChange={onChange} />

          <label htmlFor="post">POST</label>
          <input type="radio"
            name="method"
            value="POST"
            checked={method === 'POST'}
            onChange={onChange} />

          <label htmlFor="put">PUT</label>
          <input type="radio"
            name="method"
            value="PUT"
            checked={method === 'PUT'}
            onChange={onChange} />

          <label htmlFor="delete">DELETE</label>
          <input type="radio"
            name="method"
            value="DELETE"
            checked={method === 'DELETE'}
            onChange={onChange} />
        </div>

        <button>Hit it</button>

        <textarea
          placeholder="Raw JSON"
          name="body"
          value={body}
          onChange={onChange}></textarea>
      </form>
    </>
  );
};
Input.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired
};
export default Input;
