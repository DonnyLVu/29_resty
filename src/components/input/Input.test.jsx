import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Input from './Input';

describe('Input component', () => {
  afterEach(() => cleanup());
  it('renders Input', () => {
    const { asFragment } = render(
      <Input
        url=""
        method=""
        json=""
        onSubmit={() => { }}
        onChange={jest.fn()}
      />);
    expect(asFragment()).toMatchSnapshot();
  });
});
