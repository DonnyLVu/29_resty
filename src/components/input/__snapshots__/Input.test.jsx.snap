// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders Input 1`] = `
<DocumentFragment>
  <form
    class="Input"
  >
    <input
      name="url"
      placeholder="This is the URL"
      type="text"
      value=""
    />
    <div>
      <label
        data-testid="get"
      >
        GET
      </label>
      <input
        name="method"
        type="radio"
        value="GET"
      />
      <label
        for="post"
      >
        POST
      </label>
      <input
        name="method"
        type="radio"
        value="POST"
      />
      <label
        for="put"
      >
        PUT
      </label>
      <input
        name="method"
        type="radio"
        value="PUT"
      />
      <label
        for="delete"
      >
        DELETE
      </label>
      <input
        name="method"
        type="radio"
        value="DELETE"
      />
    </div>
    <button>
      Hit it
    </button>
    <textarea
      name="body"
      placeholder="Raw JSON"
    />
  </form>
</DocumentFragment>
`;
